### DISCUSSION TITLE
20240905 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#106] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  -  
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust: Release v3.0.0-beta

- https://github.com/torrust/torrust-tracker/releases/tag/v3.0.0-beta
- https://github.com/torrust/torrust-index/releases/tag/v3.0.0-beta

Final v3.0.0 will be on 30/09/2024.

### Torrust: Hash2Torrent

This week, I decided to implement a new service to obtain the torrent file from the infohash. 

- Repo: https://github.com/torrust/torrust-hash2torrent
- Live demo: https://hash2torrent.com/

I've used the BitTorrent client [rqbit](https://github.com/ikatson/rqbit).

I wanted:

- Try this BitTorrent client to see how we could integrate a BitTorrent client into the whole solution in the future. Some of Index's features may require a client. For example, we could allow users to upload a new torrent only with the infohash. The Index can use this service to get the metadata (torrent file with only the info dictionary). The index could also send a command to the client to download the torrent, for example, to classify the torrent automatically depending on the contents or even report the torrent if the content doesn't follow the Index policies.
- Research about running a client only to be used via an API and potential problems with setting up the client.

I also review the containerization of Rust apps, deployments, and basic workflows (testing, container).

@da2ce7 I also realized I could have used the BitTorrent Infrastructure Project you have been working on recently:

https://github.com/torrust/bittorrent-infrastructure-project/tree/develop/examples/get_metadata

I like how that project is organized in tiny reusable packages. For this hash2torrent project, it would have been better to use those packages instead of a full client. However, I wanted to use a full client because, in the future, it would be nice to integrate a client into the solution for the reasons I mentioned above. Besides, the development of that client is active. @ikatson even helped me to implement this project by adding a new feature to the lib (and API, although I decided to use the lib instead of the client API. I didn't want to expose a full API to use one endpoint and it seems there is no authentication):

https://github.com/ikatson/rqbit/issues/177

Thanks, @ikatson, for sharing the project and making that change.

**Comment by mario-nt** [comment]
### Index
	
- [https://github.com/torrust/torrust-index/issues/615 - tests ](https://github.com/torrust/torrust-index/pull/718/)(Torrent context and proxy context)

### Training

- The Rust programming language book
